core/src/thrift/tachyon.thrift

```diff
@@ -257,23 +287,69 @@ service WorkerService {
   bool asyncCheckpoint(1: i32 fileId)
     throws (1: TachyonException e)
 
+  /**
+   * Used to cache a block into Tachyon's cache space, worker will move the temporary block file
+   * from user folder to data folder, and update the space usage information related. then update
+   * the block information to master. 
+   */
   void cacheBlock(1: i64 userId, 2: i64 blockId)
     throws (1: FileDoesNotExistException eP, 2: SuspectedFileSizeException eS,
       3: BlockInfoException eB)
 
-  string getDataFolder()
+  /**
+   * Used to cancel a block which is being written. worker will delete the temporary block file and
+   * the location and space information related, then reclaim space allocated to the block.
+   */
+  void cancelBlock(1: i64 userId, 2: i64 blockId)
 
-  string getUserTempFolder(1: i64 userId)
+  /**
+   * Used to allocate location and space for a new coming block, worker will choose the appropriate
+   * storage directory which fits the initial block size by some allocation strategy, and the
+   * temporary file path of the block file will be returned. if there is no enough space on Tachyon
+   * storage OutOfSpaceException will be thrown, if the file is already being written by the user,
+   * FileAlreadyExistException will be thrown.
+   */
+  string getBlockLocation(1: i64 userId, 2: i64 blockId, 3: i64 initialBytes)
```

Comment:
Since this is used to request space for new coming block, how about renaming it to be "requestBlockLocation"